astropy/io/ascii/cds.py

```diff
@@ -126,7 +126,8 @@ def get_cols(self, lines):
                 col.type = self.get_col_type(col)
 
                 match = re.match(
-                    r'\? (?P<equal> =)? (?P<nullval> \S*) (\s+ (?P<descriptiontext> \S.*))?',
+                    r'(?P<order>[\[\]] \S* [\[\]])? \? (?P<equal> =)?'
```

Comment:
One last comment: I'll name this "limits" instead or "order" because the spec calls is "limits". Order comes after the null values and is "+=", "+", "-", or "-=".
```suggestion
                    r'(?P<limits>[\[\]] \S* [\[\]])? \? (?P<equal> =)?'
```